build/testing/helpers.go

```diff
@@ -0,0 +1,133 @@
+package testing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/google/go-cmp/cmp"
+)
+
+func flipt(args ...string) []string {
+	return append([]string{"/flipt"}, args...)
+}
+
+func sh(cmd string) []string {
+	return []string{"sh", "-c", cmd}
+}
+
+type stringAssertion interface {
+	assert(string) error
+}
+
+type assertConf struct {
+	success bool
+	stdout  []stringAssertion
+	stderr  []stringAssertion
+}
+
+type assertOption func(*assertConf)
+
+func fails(c *assertConf) { c.success = false }
+
+func stdout(a stringAssertion) assertOption {
+	return func(c *assertConf) {
+		c.stdout = append(c.stdout, a)
+	}
+}
+
+func stderr(a stringAssertion) assertOption {
+	return func(c *assertConf) {
+		c.stderr = append(c.stderr, a)
+	}
+}
+
+type equals string
+
+func (e equals) assert(t string) error {
+	if diff := cmp.Diff(string(e), t); diff != "" {
+		return fmt.Errorf("unexpected output: diff (-/+):\n%s", diff)
+	}
+
+	return nil
+}
+
+type contains string
+
+func (c contains) assert(t string) error {
+	if !strings.Contains(t, string(c)) {
+		return fmt.Errorf("unexpected output: %q does not contain %q", t, c)
+	}
+
+	return nil
+}
+
+type matches string
+
+func (m matches) assert(t string) error {
+	r := regexp.MustCompile(string(m))
+	if !r.MatchString(t) {
+		return fmt.Errorf("unexpected output %q does not match %q", t, m)
+	}
+
+	return nil
+}
+
+func assertExec(ctx context.Context, flipt *dagger.Container, args []string, opts ...assertOption) (*dagger.Container, error) {
+	conf := assertConf{success: true}
+	for _, opt := range opts {
+		opt(&conf)
+	}
+
+	container, err := flipt.WithExec(args).Sync(ctx)
+
+	var (
+		stdout = container.Stdout
+		stderr = container.Stderr
```

Comment:
Wdyt about renaming these variables to signify that this is coming from the container? The only reason I ask is because of the `stdout`, `stderr` functions above that accept assertions. So down below it took me some time to understand which function was which.